educator/src/Dscp/Educator/Web/Server.hs

```diff
@@ -68,19 +69,24 @@ mkStudentApiServer nat botSwitch = do
   where
     getServer handlers = hoistServerWithContext
         studentAPI
-        (Proxy :: Proxy '[GetStudentsAction])
+        (Proxy :: Proxy '[StudentCheckAction])
         nat
         (toServant handlers)
 
--- This is a temporary function that provides a dummy GetStudentAction
-createGetStudentsAction :: IO GetStudentsAction
-createGetStudentsAction = do
-    (tvr :: TVar [PublicKey]) <- atomically $ newTVar []
-    let addKeyWithSeed n =
-            let pk = snd $ withIntSeed n keyGen
-            in atomically $ modifyTVar' tvr (pk:)
-    traverse_ addKeyWithSeed [1000..1100]
-    return . GetStudentsAction $ atomically . readTVar $ tvr
+-- | Create an action which checks whether or not the
+-- student is a valid API user.
+-- If bot is enabled, all students are allowed to use API.
+createStudentCheckAction
+    :: forall ctx m. EducatorWorkMode ctx m
+    => EducatorBotSwitch
+    -> m StudentCheckAction
+createStudentCheckAction (EducatorBotOn _) =
+    return . StudentCheckAction . const $ pure True
```

Comment:
Maybe we can add student here instead of modifying each endpoint to do that?

Reply:
good point...

Reply:
Ah, no, I'd better not do it, it will break down all the tests on Educator.

Reply:
But we have only 3 tests with bot :thinking:

Reply:
Let's not risk bot correctness without necessity until we ship release. Task for the future: https://issues.serokell.io/issue/DSCP-289